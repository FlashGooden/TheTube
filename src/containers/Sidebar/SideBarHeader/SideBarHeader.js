import React from 'react'
import {Menu} from 'semantic-ui-react'
import './SideBarHeader.scss'

export function SideBarHeader(props) {
  const heading = props.title ? props.title.toUpperCase() : '';  
  return (
    <Menu.item>
      <Menu.Header className="side-bar-header">{heading}</Menu.Header>
    </Menu.item>
  )
}
