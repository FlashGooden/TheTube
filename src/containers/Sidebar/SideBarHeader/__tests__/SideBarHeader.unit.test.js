import React from "react";
import { shallow } from "enzyme";
import { SideBarHeader } from "../SideBarHeader";
import { Menu } from "semantic-ui-react";

describe("SideBarHeader", () => {
  test("should render SidebarHeader", () => {
    const wrapper = shallow(<SideBarHeader />);
    expect(wrapper).toMatchSnapshot();
  });

  test("renders with empty props.title", () => {
    const wrapper = shallow(<SideBarHeader title="" />);
    expect(wrapper).toMatchSnapshot();
  });
  test("renders with title passed", () => {
    const wrapper = shallow(<SideBarHeader title="Subscriptions" />);
    expect(wrapper).toMatchSnapshot();
  });
});
