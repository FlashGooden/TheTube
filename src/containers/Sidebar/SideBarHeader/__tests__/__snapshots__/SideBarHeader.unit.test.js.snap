// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBarHeader renders with empty props.title 1`] = `
<MenuItem>
  <MenuHeader
    className="side-bar-header"
  />
</MenuItem>
`;

exports[`SideBarHeader renders with title passed 1`] = `
<MenuItem>
  <MenuHeader
    className="side-bar-header"
  >
    SUBSCRIPTIONS
  </MenuHeader>
</MenuItem>
`;

exports[`SideBarHeader should render SidebarHeader 1`] = `
<MenuItem>
  <MenuHeader
    className="side-bar-header"
  />
</MenuItem>
`;
